components/web_discovery/browser/double_fetcher.cc

```diff
@@ -0,0 +1,132 @@
+#include "brave/components/web_discovery/browser/util.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+
+namespace web_discovery {
+
+namespace {
+constexpr char kUrlKey[] = "url";
+constexpr char kAssociatedDataKey[] = "assoc_data";
+
+constexpr base::TimeDelta kRequestMaxAge = base::Hours(1);
+constexpr base::TimeDelta kMinRequestInterval =
+    base::Minutes(1) - base::Seconds(5);
+constexpr base::TimeDelta kMaxRequestInterval =
+    base::Minutes(1) + base::Seconds(5);
+constexpr size_t kMaxRetries = 3;
+constexpr size_t kMaxDoubleFetchResponseSize = 2 * 1024 * 1024;
+
+constexpr net::NetworkTrafficAnnotationTag kFetchNetworkTrafficAnnotation =
+    net::DefineNetworkTrafficAnnotation("wdp_doublefetch", R"(
+    semantics {
+      sender: "Brave Web Discovery Double Fetch"
+      description:
+        "Retrieves a page of interest without session cookies for
+         scraping and reporting via Web Discovery."
+      trigger:
+        "Requests are sent minutes after the original
+         page request is made by the user."
+      data: "Page data"
+      destination: WEBSITE
+    }
+    policy {
+      cookies_allowed: NO
+      setting:
+        "Users can opt-in or out via brave://settings/search"
+    })");
+
+}  // namespace
+
+DoubleFetcher::DoubleFetcher(
+    PrefService* profile_prefs,
+    network::SharedURLLoaderFactory* shared_url_loader_factory,
+    FetchedCallback callback)
+    : profile_prefs_(profile_prefs),
+      shared_url_loader_factory_(shared_url_loader_factory),
+      request_queue_(profile_prefs,
+                     kScheduledDoubleFetches,
+                     kRequestMaxAge,
+                     kMinRequestInterval,
+                     kMaxRequestInterval,
+                     kMaxRetries,
+                     base::BindRepeating(&DoubleFetcher::OnFetchTimer,
+                                         base::Unretained(this))),
+      callback_(callback) {}
+
+DoubleFetcher::~DoubleFetcher() = default;
+
+void DoubleFetcher::ScheduleDoubleFetch(const GURL& url,
+                                        base::Value associated_data) {
+  base::Value::Dict fetch_dict;
+  fetch_dict.Set(kUrlKey, url.spec());
+  fetch_dict.Set(kAssociatedDataKey, std::move(associated_data));
+
+  request_queue_.ScheduleRequest(base::Value(std::move(fetch_dict)));
+}
+
+void DoubleFetcher::OnFetchTimer(const base::Value& request_data) {
+  const auto* fetch_dict = request_data.GetIfDict();
+  const auto* url_str = fetch_dict ? fetch_dict->FindString(kUrlKey) : nullptr;
+  if (!url_str) {
+    request_queue_.NotifyRequestComplete(true);
+    return;
+  }
+
+  GURL url(*url_str);
+  auto resource_request = CreateResourceRequest(url);
+  url_loader_ = network::SimpleURLLoader::Create(
```

Comment:
Does this actually set `network::mojom::CredentialsMode::kOmit`? I don't see where it drops the cookies.

Reply:
https://github.com/brave/brave-core/blob/a9d3f3bf02800afaeac581e905eac7f014fc4498/components/web_discovery/browser/util.cc#L47-L52

Reply:
Thanks, I missed it while I was going through.